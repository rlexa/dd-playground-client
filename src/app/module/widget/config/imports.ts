import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
