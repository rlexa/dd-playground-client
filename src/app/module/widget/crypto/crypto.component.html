<div ddFlexbox="hyscsw">
  <mat-card class="childhv">
    <mat-card-title>Actions: API</mat-card-title>
    <mat-card-content>
      <div ddFlexbox="vys">
        <h1 class="color-warn">CORS/PROXY KILLS API.</h1>

        <mat-list class="mat-list-auto-height">
          <h3 mat-subheader>Ping</h3>
          <mat-list-item>
            <div ddFlexbox="hyc">
              <button mat-button (click)="wsBlockchainPing()">Blockchain WS</button>
              <button mat-button (click)="wsBlockCypherPing()">BlockCypher WS</button>
            </div>
          </mat-list-item>

          <h3 mat-subheader>Address</h3>
          <mat-list-item>
            <div ddFlexbox="hyc">
              <mat-form-field class="childh value-numeric">
                <input
                  matInput
                  [ngModel]="bitcoinAddress$ | async"
                  (ngModelChange)="bitcoinAddress$.next($event)"
                  placeholder="Bitcoin Address"
                />
              </mat-form-field>
              <div ddFlexbox="vys">
                <button mat-button (click)="apiBlockchainBitcoinAddress()">Blockchain API</button>
                <button mat-button (click)="wsBlockchainAddressLatestTransaction()">Blockchain WS</button>
              </div>
              <div ddFlexbox="vys">
                <button mat-button (click)="apiBlockCypherBitcoinAddress()">BlockCypher API</button>
                <button mat-button (click)="wsBlockCypherAddressConfirmedTx()">BlockCypher WS</button>
              </div>
            </div>
          </mat-list-item>

          <h3 mat-subheader>Stats</h3>
          <mat-list-item>
            <div ddFlexbox="hyc">
              <div ddFlexbox="vys">
                <button mat-button (click)="apiWebBitcoinStats()">WebBitcoin API</button>
              </div>
            </div>
          </mat-list-item>
        </mat-list>
      </div>
    </mat-card-content>
  </mat-card>

  <mat-card class="childhv">
    <mat-card-title>Actions: Output</mat-card-title>
    <mat-card-content>
      <app-simple-view [data]="anyData$ | async" [isWidthConstrained]="false" [isDense]="true"></app-simple-view>
    </mat-card-content>
  </mat-card>
</div>
