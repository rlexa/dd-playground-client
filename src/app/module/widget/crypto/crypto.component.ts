import {ChangeDetectionStrategy, Component, OnDestroy, OnInit} from '@angular/core';
import {rxNext_} from 'dd-rxjs';
import {BehaviorSubject, of, Subject} from 'rxjs';
import {switchMap} from 'rxjs/operators';
import {cleanupRx} from 'src/app/util/cleanup-rx';
import {CryptoApiService} from '../../service/crypto-api';

@Component({
  selector: 'app-crypto',
  templateUrl: './crypto.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CryptoComponent implements OnDestroy, OnInit {
  constructor(private readonly api: CryptoApiService) {}

  readonly bitcoinAddress$ = new BehaviorSubject('1BoatSLRHtKNngkdXEeobR76b53LETtpyT');
  readonly anyData$ = new Subject();

  ngOnDestroy() {
    cleanupRx(this.anyData$, this.bitcoinAddress$);
  }
  ngOnInit() {}

  apiBlockchainBitcoinAddress = () =>
    of(this.bitcoinAddress$.value || null)
      .pipe(switchMap((_) => this.api.blockchainSingleAddress$({address: _})))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));

  apiBlockCypherBitcoinAddress = () =>
    of(this.bitcoinAddress$.value || null)
      .pipe(switchMap((_) => this.api.blockcypherSingleAddress$({address: _})))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));

  apiWebBitcoinStats = () =>
    of(this.bitcoinAddress$.value || null)
      .pipe(switchMap((_) => this.api.webbitcoinStats$()))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));

  wsBlockchainPing = () =>
    of(null)
      .pipe(switchMap((_) => this.api.wsBlockchainPing$()))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));

  wsBlockchainAddressLatestTransaction = () =>
    of(this.bitcoinAddress$.value || null)
      .pipe(switchMap((_) => this.api.wsBlockchainAddressLatestTransaction$(_)))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));

  wsBlockCypherPing = () =>
    of(null)
      .pipe(switchMap((_) => this.api.wsBlockCypherPing$()))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));

  wsBlockCypherAddressConfirmedTx = () =>
    of(this.bitcoinAddress$.value || null)
      .pipe(switchMap((_) => this.api.wsBlockCypherAddressConfirmedTx$(_)))
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));
}
