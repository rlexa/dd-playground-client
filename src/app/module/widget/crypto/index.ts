export * from './crypto.component';
export * from './crypto.module';
