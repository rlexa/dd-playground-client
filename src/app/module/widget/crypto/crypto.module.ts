import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ROUTE_ROOT, ROUTE_WILDCARD } from 'app/routing';
import { CryptoComponent } from './crypto.component';
import { imports } from './imports';

const ROUTING = [
  { path: ROUTE_ROOT, component: CryptoComponent },
  { path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT, pathMatch: 'full' },
];

@NgModule({
  imports: [...imports, RouterModule.forChild(ROUTING)], exports: [CryptoComponent], declarations: [CryptoComponent]
})
export class CryptoModule { }
