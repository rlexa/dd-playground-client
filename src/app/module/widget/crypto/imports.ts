import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
