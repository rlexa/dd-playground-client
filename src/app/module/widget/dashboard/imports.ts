import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
