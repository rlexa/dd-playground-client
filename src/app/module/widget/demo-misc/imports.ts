import { CommonModule } from '@angular/common';
import { MatButtonModule, MatCardModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { GraphskyApiModule } from 'app/module/service/graphsky-api';
import { DiagramPolynomModule } from 'app/module/widget/diagram-polynom';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  FlexboxModule,
  DiagramPolynomModule,
  GraphskyApiModule,
  SimpleViewModule,
];
