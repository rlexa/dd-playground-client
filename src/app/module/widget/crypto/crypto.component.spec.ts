import {async, TestBed} from '@angular/core/testing';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';
import {RouterTestingModule} from '@angular/router/testing';
import {CryptoComponent} from './crypto.component';
import {imports} from './imports';

describe('CryptoComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [NoopAnimationsModule, RouterTestingModule, ...imports],
      declarations: [CryptoComponent],
      providers: [],
    }).compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(CryptoComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
