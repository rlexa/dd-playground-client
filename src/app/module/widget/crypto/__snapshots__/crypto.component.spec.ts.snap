// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoComponent is created 1`] = `
<app-crypto
  anyData$={[Function Subject]}
  api={[Function CryptoApiService]}
  apiBlockCypherBitcoinAddress={[Function Function]}
  apiBlockchainBitcoinAddress={[Function Function]}
  bitcoinAddress$={[Function BehaviorSubject]}
  done$={[Function DoneSubject]}
  wsBlockCypherAddressConfirmedTx={[Function Function]}
  wsBlockCypherPing={[Function Function]}
  wsBlockchainAddressLatestTransaction={[Function Function]}
  wsBlockchainPing={[Function Function]}
>
  <div
    ddflexbox="hyscsw"
    ng-reflect-dd-flexbox="hyscsw"
    style="display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: flex-start; flex-wrap: wrap;"
  >
    <mat-card
      class="childhv mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Actions: API
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <div
          ddflexbox="vys"
          ng-reflect-dd-flexbox="vys"
          style="display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
        >
          <h1
            class="color-warn"
          >
            CORS/PROXY KILLS API.
          </h1>
          <mat-list
            class="mat-list-auto-height mat-list"
          >
            <h3
              class="mat-subheader"
              mat-subheader=""
            >
              Ping
            </h3>
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <div
                  ddflexbox="hyc"
                  ng-reflect-dd-flexbox="hyc"
                  style="display: flex; flex-direction: row; justify-content: flex-start; align-items: center; align-content: stretch; flex-wrap: no-wrap;"
                >
                  <button
                    class="mat-button _mat-animation-noopable"
                    mat-button=""
                  >
                    <span
                      class="mat-button-wrapper"
                    >
                      Blockchain WS
                    </span>
                    <div
                      class="mat-button-ripple mat-ripple"
                      matripple=""
                      ng-reflect-centered="false"
                      ng-reflect-disabled="false"
                      ng-reflect-trigger="[object HTMLButtonElement]"
                    />
                    <div
                      class="mat-button-focus-overlay"
                    />
                  </button>
                  <button
                    class="mat-button _mat-animation-noopable"
                    mat-button=""
                  >
                    <span
                      class="mat-button-wrapper"
                    >
                      BlockCypher WS
                    </span>
                    <div
                      class="mat-button-ripple mat-ripple"
                      matripple=""
                      ng-reflect-centered="false"
                      ng-reflect-disabled="false"
                      ng-reflect-trigger="[object HTMLButtonElement]"
                    />
                    <div
                      class="mat-button-focus-overlay"
                    />
                  </button>
                </div>
              </div>
            </mat-list-item>
            <h3
              class="mat-subheader"
              mat-subheader=""
            >
              Address
            </h3>
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <div
                  ddflexbox="hyc"
                  ng-reflect-dd-flexbox="hyc"
                  style="display: flex; flex-direction: row; justify-content: flex-start; align-items: center; align-content: stretch; flex-wrap: no-wrap;"
                >
                  <mat-form-field
                    class="childh value-numeric mat-form-field ng-tns-c3-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
                  >
                    <div
                      class="mat-form-field-wrapper"
                    >
                      <div
                        class="mat-form-field-flex"
                      >
                        
                        
                        <div
                          class="mat-form-field-infix"
                        >
                          <input
                            aria-invalid="false"
                            aria-required="false"
                            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                            id="mat-input-0"
                            matinput=""
                            ng-reflect-model="1BoatSLRHtKNngkdXEeobR76b53LET"
                            ng-reflect-placeholder="Bitcoin Address"
                            placeholder="Bitcoin Address"
                          />
                          <span
                            class="mat-form-field-label-wrapper"
                          >
                            
                            <label
                              aria-owns="mat-input-0"
                              class="mat-form-field-label ng-tns-c3-0 mat-empty mat-form-field-empty ng-star-inserted"
                              for="mat-input-0"
                              id="mat-form-field-label-1"
                              ng-reflect-ng-switch="false"
                            >
                              
                              
                              Bitcoin Address
                              
                              
                            </label>
                          </span>
                        </div>
                        
                      </div>
                      
                      <div
                        class="mat-form-field-underline ng-tns-c3-0 ng-star-inserted"
                      >
                        <span
                          class="mat-form-field-ripple"
                        />
                      </div>
                      <div
                        class="mat-form-field-subscript-wrapper"
                        ng-reflect-ng-switch="hint"
                      >
                        
                        
                        <div
                          class="mat-form-field-hint-wrapper ng-tns-c3-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                        >
                          
                          <div
                            class="mat-form-field-hint-spacer"
                          />
                        </div>
                      </div>
                    </div>
                  </mat-form-field>
                  <div
                    ddflexbox="vys"
                    ng-reflect-dd-flexbox="vys"
                    style="display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
                  >
                    <button
                      class="mat-button _mat-animation-noopable"
                      mat-button=""
                    >
                      <span
                        class="mat-button-wrapper"
                      >
                        Blockchain API
                      </span>
                      <div
                        class="mat-button-ripple mat-ripple"
                        matripple=""
                        ng-reflect-centered="false"
                        ng-reflect-disabled="false"
                        ng-reflect-trigger="[object HTMLButtonElement]"
                      />
                      <div
                        class="mat-button-focus-overlay"
                      />
                    </button>
                    <button
                      class="mat-button _mat-animation-noopable"
                      mat-button=""
                    >
                      <span
                        class="mat-button-wrapper"
                      >
                        Blockchain WS
                      </span>
                      <div
                        class="mat-button-ripple mat-ripple"
                        matripple=""
                        ng-reflect-centered="false"
                        ng-reflect-disabled="false"
                        ng-reflect-trigger="[object HTMLButtonElement]"
                      />
                      <div
                        class="mat-button-focus-overlay"
                      />
                    </button>
                  </div>
                  <div
                    ddflexbox="vys"
                    ng-reflect-dd-flexbox="vys"
                    style="display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
                  >
                    <button
                      class="mat-button _mat-animation-noopable"
                      mat-button=""
                    >
                      <span
                        class="mat-button-wrapper"
                      >
                        BlockCypher API
                      </span>
                      <div
                        class="mat-button-ripple mat-ripple"
                        matripple=""
                        ng-reflect-centered="false"
                        ng-reflect-disabled="false"
                        ng-reflect-trigger="[object HTMLButtonElement]"
                      />
                      <div
                        class="mat-button-focus-overlay"
                      />
                    </button>
                    <button
                      class="mat-button _mat-animation-noopable"
                      mat-button=""
                    >
                      <span
                        class="mat-button-wrapper"
                      >
                        BlockCypher WS
                      </span>
                      <div
                        class="mat-button-ripple mat-ripple"
                        matripple=""
                        ng-reflect-centered="false"
                        ng-reflect-disabled="false"
                        ng-reflect-trigger="[object HTMLButtonElement]"
                      />
                      <div
                        class="mat-button-focus-overlay"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </mat-list-item>
          </mat-list>
        </div>
      </mat-card-content>
    </mat-card>
    <mat-card
      class="childhv mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Actions: Output
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <app-simple-view
          class="ng-tns-c4-1"
          ng-reflect-is-dense="true"
          ng-reflect-is-width-constrained="false"
        >
          
        </app-simple-view>
      </mat-card-content>
    </mat-card>
  </div>
</app-crypto>
`;
